packages/contracts-bedrock/test/L2/OptimismSuperchainERC20.t.sol

```diff
@@ -0,0 +1,384 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+// Testing utilities
+import { Test } from "forge-std/Test.sol";
+
+// Libraries
+import { Predeploys } from "src/libraries/Predeploys.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { stdStorage, StdStorage } from "forge-std/Test.sol";
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import { IL2ToL2CrossDomainMessenger } from "src/L2/IL2ToL2CrossDomainMessenger.sol";
+import { ERC1967Proxy } from "@openzeppelin/contracts/proxy/ERC1967/ERC1967Proxy.sol";
+import { Initializable } from "@openzeppelin/contracts-v5/proxy/utils/Initializable.sol";
+
+// Target contract
+import {
+    OptimismSuperchainERC20,
+    IOptimismSuperchainERC20Extension,
+    CallerNotL2ToL2CrossDomainMessenger,
+    InvalidCrossDomainSender,
+    OnlyBridge,
+    ZeroAddress
+} from "src/L2/OptimismSuperchainERC20.sol";
+import { ISuperchainERC20Extensions } from "src/L2/ISuperchainERC20.sol";
+
+/// @title OptimismSuperchainERC20Test
+/// @notice Contract for testing the OptimismSuperchainERC20 contract.
+contract OptimismSuperchainERC20Test is Test {
+    address internal constant ZERO_ADDRESS = address(0);
+    address internal constant REMOTE_TOKEN = address(0x123);
+    string internal constant NAME = "OptimismSuperchainERC20";
+    string internal constant SYMBOL = "SCE";
+    uint8 internal constant DECIMALS = 18;
+    address internal constant BRIDGE = Predeploys.L2_STANDARD_BRIDGE;
+    address internal constant MESSENGER = Predeploys.L2_TO_L2_CROSS_DOMAIN_MESSENGER;
+
+    OptimismSuperchainERC20 public superchainERC20Impl;
+    OptimismSuperchainERC20 public superchainERC20;
+
+    /// @notice Sets up the test suite.
+    function setUp() public {
+        superchainERC20Impl = new OptimismSuperchainERC20();
+        superchainERC20 = _deploySuperchainERC20Proxy(REMOTE_TOKEN, NAME, SYMBOL, DECIMALS);
+    }
+
+    /// @notice Helper function to deploy a proxy of the OptimismSuperchainERC20 contract.
+    function _deploySuperchainERC20Proxy(
+        address _remoteToken,
+        string memory _name,
+        string memory _symbol,
+        uint8 _decimals
+    )
+        internal
+        returns (OptimismSuperchainERC20)
+    {
+        return OptimismSuperchainERC20(
+            address(
+                new ERC1967Proxy(
```

Comment:
Longer term we may want to have tests where we use the same proxy that we plan on using in production